automated-testing/performance-testing/load-testing.md

```diff
@@ -0,0 +1,69 @@
+
+## Applying the Load Testing
+
+### Planning
+
+1. **Identify key scenarios to measure**  - Gather these scenarios from Product Owner, they should provide a representative sample of real world traffic.
+2. **Determine expected normal and peak load for the scenarios** -  Determine a load level such as concurrent users or requests per second to find the size of the load test you will run.
+3. **Identify success criteria metrics** - These may be on testing side such as response time and error rate, or they may be on the system side such as CPU and memory usage.
+
+### Execution
+
+It is recommended to use an existing testing framework (see below). These tools will provide a method of both specifying the user activity scenarios as well as how to execute those at load. It is common to slowly ramp up to your desired load just to be safe. Once you have reached your defined workload.
+
+Before starting please be aware of any restrictions on your network such as DDOS protection where you may need to notify a network administrator or apply for an exemption.
+
+### Further Testing
+
+After completing your load test you should be set up to continue on to additional related testing such as;
+
+- **Soak Testing** - Performing a load test over an extended period of time to ensure long term stability.
+- **Stress Testing** - Gradually increasing the load to find the limits of the system and identify the maximum capacity.
+- **Spike Testing** - Introduce a sharp short-term increase into the load scenarios.
+- **Scalability Testing** - Re-testing of a system as your expand horizontally or vertically to measure how it scales.
+
+## Load Testing Frameworks and Tools
+
+Here are a few popular load testing frameworks you may consider and the languages used to define your scenarios.
+
+- **JMeter** (<https://jmeter.apache.org/>) - Has built in patterns to test without coding, but can be extended with Java.
+- **Artillery** (<https://artillery.io/>) - Write your scenarios in Javascript, executes a node application.
+- **Gatling** (<https://gatling.io/>) -  Write your scenarios in Scala with their DSL.
+- **Locust** (<https://locust.io/>) - Write your scenarios in Python using the concept of concurrent user activity.
+- **K6** (<https://k6.io/>) - Write your test scenarios in Javascript, available as open source or as SaaS.
+
+## Conclusion
+
+A load test is critical step to understand if a target system will be reliable under the expected real world traffic.
+
+Of course its only as good as your ability to predict the expected load, so its important to follow up with other further testing to truly understand how your system behaves in different situations.
+
+## Resources
+
+List additional readings about this test type for those that would like to dive deeper.
+
+- [https://docs.microsoft.com/en-us/azure/architecture/framework/scalability/load-testing](https://docs.microsoft.com/en-us/azure/architecture/framework/scalability/load-testing)

```

Comment:
You may want to use `Load Testing`, page title, instead of url for easier understanding